src/main/scala/com/hungry/cars/http/routes/UserRoutes.scala

```diff
@@ -0,0 +1,39 @@
+package com.hungry.cars.http.routes
+
+import cats.data._
+import cats.effect.ContextShift
+import cats.effect.IO
+import cats.effect.Timer
+import com.hungry.cars.domain.ValidatedCreateUserRequest
+import com.hungry.cars.domain.error.UserError
+import com.hungry.cars.domain.error.UserError.userErrorEncoder
+import com.hungry.cars.http.in.CreateUserRequest
+import com.hungry.cars.http.in.CreateUserRequest.createUserDecoder
+import com.hungry.cars.services.UserService
+import org.http4s.EntityDecoder
+import org.http4s.HttpRoutes
+import org.http4s.circe.CirceEntityCodec.circeEntityEncoder
+import org.http4s.circe._
+import org.http4s.dsl.io._
+
+class UserRoutes(userService: UserService)(implicit cs: ContextShift[IO], timer: Timer[IO]) {
+
+  implicit private val createUserRequestDecoder: EntityDecoder[IO, CreateUserRequest] = jsonOf[IO, CreateUserRequest]
+
+  val routes: HttpRoutes[IO] = HttpRoutes
+    .of[IO] {
+
+      case req @ POST -> Root / "users" =>
```

Comment:
Ten path sugeruje, że ktoś tworzy tych użytkowników. Może to powinno być "users" / "register"?